frontend/catalyst/compiled_functions.py

```diff
@@ -167,6 +168,8 @@ def _exec(shared_object, has_return, out_type, numpy_dict, *args):
         if out_type is not None:
             keep_outputs = [k for _, k in out_type]
             retval = [r for (k, r) in zip(keep_outputs, retval) if k]
+
+        retval = [jnp.asarray(arr) for arr in retval]
```

Comment:
@erick-xanadu is `retval` always guaranteed to be a NumPy array? I'm just thinking of qjit functions that return arbitrary pytrees/lists, etc. We want to ensure that the type the user chooses to return is preserved

Reply:
No, here it is the flat output.

Reply:
Just clarifying above:

> @erick-xanadu is retval always guaranteed to be a NumPy array?

Yes.

> I'm just thinking of qjit functions that return arbitrary pytrees/lists, etc. 

Here it is the flat output.

Reply:
Sorry, I just realized I have a typo: I meant to say 

> @erick-xanadu is `arr` always guaranteed to be a NumPy array?

Reply:
Yes.